File: webapp/Locust/Scripts/app/services/TestExecutionService.js

(function () {
    'use strict';

    var app = angular.module(appName);

    app.service('TestExecutionService', ['$resource', function ($resource) {

        var noop = angular.noop;
        var resource = $resource('/api/TestExecution/:action/:id',
            { id: '@Id' }, {
                Save: {
                    method: 'POST'
                },
                Get: {
                    method: 'GET',
                    isArray: false
                },
                Update: {
                    method: 'PUT'
                },
                GetForGroup: {
                    method: 'GET',
                    isArray: true
                },
                ChangeState: {
                    method: 'GET',
                    isArray: false
                },
                GetByProject: {
                    method: 'GET',
                    isArray: true
                }
            });
        self = this;



        self.Save = function (entity, success, error) {
            return resource.Save({ action: "Save" }, entity, success || noop, error || noop);
        };

        self.Update = function (entity, success, error) {
            return resource.Update({ action: "Update" }, entity, success || noop, error || noop);
        };

        self.Get = function (id, success, error) {
            return resource.Get({ action: "Get", id: id }, success || noop, error || noop);
        };

        self.GetForGroup = function (executionId, success, error) {
            return resource.GetForGroup({ action: "GetForGroup", executionId: executionId }, success || noop, error || noop);
        };

        self.ChangeState = function (id,state ,success, error) {
            return resource.ChangeState({ action: "ChangeState", id: id, state: state }, success || noop, error || noop);
        };

        self.GetByProject = function (projectId,success, error) {
            return resource.GetByProject({ action: "GetByProject", projectId: projectId}, success || noop, error || noop);
        };

    }]);

})(angular);

File: webapp/Locus.Core/Services/TestExecutionService.cs
using Locus.Core.Models;
using Locus.Core.Repositories;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Locus.Core.Services
{
    public interface ITestExecutionService
    {
        TestExecution Save(TestExecution testExecution);

        TestExecution Update(TestExecution testExecution);

        TestExecution ChangeState(int id,string state);

        TestExecution Get(int id);

        List<TestExecution> GetForGroup(int executionId);

        TestExecution GetLastExecution(int groupId, int projectId);

        ExecutionGroup GetByTestExecution(int groupId);

        List<TestExecution> GetByProject(int projectId);
    }

    public class TestExecutionService : ITestExecutionService
    {
        private readonly ITestExecutionRepository _testExecutionRepository;
        private readonly IExecutionGroupRepository _executionGroupRepository;

        public TestExecutionService(ITestExecutionRepository testExecutionRepository)
        {
            _testExecutionRepository = testExecutionRepository;
        }

        public TestExecutionService(ITestExecutionRepository testExecutionRepository, IExecutionGroupRepository executionGroupRepository)
        {
            _testExecutionRepository = testExecutionRepository;
            _executionGroupRepository = executionGroupRepository;
        }

        public TestExecution ChangeState(int id, string state)
        {
            return _testExecutionRepository.ChangeState(id, state);
        }

        public TestExecution Get(int id)
        {
            return _testExecutionRepository.Get(id);
        }

        public List<TestExecution> GetByProject(int projectId)
        {
            return _testExecutionRepository.GetByProject(projectId);
        }

        public ExecutionGroup GetByTestExecution(int groupId)
        {
            return _testExecutionRepository.GetByTestExecution(groupId);
        }

        public List<TestExecution> GetForGroup(int executionId)
        {
            return _testExecutionRepository.GetForGroup(executionId);
        }

        public TestExecution GetLastExecution(int groupId, int projectId)
        {
            return _testExecutionRepository.GetLastExecution(groupId, projectId);
        }

        public TestExecution Save(TestExecution testExecution)
        {
            return _testExecutionRepository.Save(testExecution);
        }

        public TestExecution Update(TestExecution testExecution)
        {
            return _testExecutionRepository.Update(testExecution);
        }
    }
}
